addons/hr_expense/hr_expense.py

```diff
@@ -388,94 +333,150 @@ def action_view_move(self, cr, uid, ids, context=None):
     _inherit = "account.move.line"
 
-    def reconcile(self, cr, uid, ids, type='auto', writeoff_acc_id=False, writeoff_period_id=False, writeoff_journal_id=False, context=None):
-        res = super(account_move_line, self).reconcile(cr, uid, ids, type=type, writeoff_acc_id=writeoff_acc_id, writeoff_period_id=writeoff_period_id, writeoff_journal_id=writeoff_journal_id, context=context)
+    @api.multi
+    def reconcile(self, type = 'auto', writeoff_acc_id = False, writeoff_period_id = False, writeoff_journal_id = False):
+        res = super(AccountMoveLine, self).reconcile(type = type, writeoff_acc_id = writeoff_acc_id, writeoff_period_id = writeoff_period_id, writeoff_journal_id = writeoff_journal_id)
         #when making a full reconciliation of account move lines 'ids', we may need to recompute the state of some hr.expense
-        account_move_ids = [aml.move_id.id for aml in self.browse(cr, uid, ids, context=context)]
-        expense_obj = self.pool.get('hr.expense.expense')
-        currency_obj = self.pool.get('res.currency')
-        if account_move_ids:
-            expense_ids = expense_obj.search(cr, uid, [('account_move_id', 'in', account_move_ids)], context=context)
-            for expense in expense_obj.browse(cr, uid, expense_ids, context=context):
-                if expense.state == 'done':
-                    #making the postulate it has to be set paid, then trying to invalidate it
-                    new_status_is_paid = True
-                    for aml in expense.account_move_id.line_id:
-                        if aml.account_id.type == 'payable' and not currency_obj.is_zero(cr, uid, expense.company_id.currency_id, aml.amount_residual):
-                            new_status_is_paid = False
-                    if new_status_is_paid:
-                        expense_obj.write(cr, uid, [expense.id], {'state': 'paid'}, context=context)
+        expenses = self.env['hr_expense.sheet'].search([('account_move_id', 'in', self.mapped("move_id").ids)])
+        for expense in expenses:
+            if expense.state == 'done':
+                #making the postulate it has to be set paid, then trying to invalidate it
+                new_status_is_paid = True
+                for aml in expense.account_move_id.line_id:
+                    if aml.account_id.type == 'payable' and not expense.company_id.currency_id.is_zero(aml.amount_residual):
+                        new_status_is_paid = False
+                if new_status_is_paid:
+                    expense.state = 'paid'
         return res
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    @api.multi
+    def _expenses_amount(self):
+        HrExpenseSheet = self.env['hr_expense.sheet']
+        for employee in self:
+            sheet = HrExpenseSheet.search([('employee_id', '=', employee.id), ('state', '=', 'confirm')], limit=1)
+            if sheet:
+                employee.expenses_amount = sheet.amount
```

Comment:
Why you search only one Expense Sheet related to an Employee ???
Generally stat buttons shows all the expense related to an Employee.